agent/src/index.ts

```diff
@@ -384,9 +384,12 @@ export async function createAgent(
         throw new Error("Invalid TEE configuration");
     }
 
-    const goatPlugin = await createGoatPlugin((secret) =>
-        getSecret(character, secret)
-    );
+    let goatPlugin: any | undefined;
+    if (getSecret(character, "ALCHEMY_API_KEY")) {
+        goatPlugin = await createGoatPlugin((secret) =>
+            getSecret(character, secret)
+        );
+    }
```

Comment:
nit: In order to not duplicate the secret check, you could do 

```suggestion
    const getGoatPlugin = async () => await createGoatPlugin((secret) =>
        getSecret(character, secret)
    );
```

and then later on line 440 do

```
            getSecret(character, "ALCHEMY_API_KEY") ? getGoatPlugin() : null,
```